src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Ak8963
+{
+    /// <summary>
+    /// AK8963 class implementing a magnetometer
+    /// </summary>
+    public class Ak8963 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private MeasurementMode _measurementMode;
+        private OutputBitMode _outputBitMode;
+        private bool _selfTest = false;
+        private Ak8963I2cBase _ak8963Interface;
+        private bool _autoDispose;
+
+        public const byte DefaultI2cAddress = 0x0C;
+
+        public Ak8963(I2cDevice i2CDevice) : this(i2CDevice, new Ak8963I2c())
+        { }
+
+        public Ak8963(I2cDevice i2CDevice, Ak8963I2cBase ak8963Interface, bool autoDispose = true)
```

Comment:
So in the case that the interface abstraction is actually needed (which I'm not yet convinced we do) I would have this constructor be internal, so that only this class and the Mpu9250 call this constructor, but a regular user would only see the default constructor which takes in an I2cdevice. That way, it is not confusing for them this implementation detail.

Reply:
The AK8963 is not in the same project neither namespace as the MPU9250. So I can't make the constructor private. The AK8963 can be used alone.
the AK8963 has a simple constructor that hide all this
```csharp
        public Ak8963(I2cDevice i2CDevice) : this(i2CDevice, new Ak8963I2c())
        { }
```